src/main/java/picard/sam/ReorderSam.java

```diff
@@ -106,69 +117,58 @@
-
-        @Override
-        public File getReferenceFile() {
-            return REFERENCE;
-        };
-    }
-
     protected int doWork() {
         IOUtil.assertFileIsReadable(INPUT);
-        IOUtil.assertFileIsReadable(REFERENCE_SEQUENCE);
+        IOUtil.assertFileIsReadable(SEQUENCE_DICTIONARY);
         IOUtil.assertFileIsWritable(OUTPUT);
 
-        final SamReader in = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(INPUT);
-
-        final ReferenceSequenceFile reference = ReferenceSequenceFileFactory.getReferenceSequenceFile(REFERENCE_SEQUENCE);
-        final SAMSequenceDictionary refDict = reference.getSequenceDictionary();
+        try (final SamReader in = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(INPUT)) {
 
-        if (refDict == null) {
-            log.error("No reference sequence dictionary found. Aborting.  You can create a sequence dictionary for the reference fasta using CreateSequenceDictionary.jar.");
-            CloserUtil.close(in);
-            return 1;
-        }
-
-        printDictionary("SAM/BAM file", in.getFileHeader().getSequenceDictionary());
-        printDictionary("Reference", refDict);
-        final Map<Integer, Integer> newOrder = buildSequenceDictionaryMap(refDict, in.getFileHeader().getSequenceDictionary());
-
-        // has to be after we create the newOrder
-        final SAMFileHeader outHeader = in.getFileHeader().clone();
-        outHeader.setSequenceDictionary(refDict);
+            final SAMSequenceDictionary outputDictionary = SAMSequenceDictionaryExtractor.extractDictionary(SEQUENCE_DICTIONARY.toPath());
 
-        log.info("Writing reads...");
-        if (in.hasIndex()) {
-            try( final SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(outHeader, true, OUTPUT)) {
+            if (outputDictionary == null) {
+                log.error("No reference sequence dictionary found. Aborting.  You can create a sequence dictionary for the reference fasta using CreateSequenceDictionary.jar.");
+                return 1;
+            }
 
-                // write the reads in contig order
-                for (final SAMSequenceRecord contig : refDict.getSequences()) {
-                    final SAMRecordIterator it = in.query(contig.getSequenceName(), 0, 0, false);
-                    writeReads(out, it, newOrder, contig.getSequenceName());
-                }
-                // don't forget the unmapped reads
-                writeReads(out, in.queryUnmapped(), newOrder, "unmapped");
+            printDictionary("SAM/BAM file", in.getFileHeader().getSequenceDictionary());
+            printDictionary("Reference", outputDictionary);
+            final Map<Integer, Integer> newOrder;
+            try {
+                newOrder = buildSequenceDictionaryMap(outputDictionary, in.getFileHeader().getSequenceDictionary());
+            } catch (PicardException e) {
+                log.error(e.getMessage());
```

Comment:
I'd just pass `e`, so the stack trace also gets logged.

Reply:
I thought we prefer not to emit stack traces if possible...but Ok.